### ISSUE TITLE
`declaration-block-trailing-semicolon` still takes a semicolon out of a `//` comment behind a custom property under Less

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#722]
Left by the branch closing [#359](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/359), which reads such a semicolon as the comment's text only behind a declaration of an ordinary property and a mixin or detached ruleset call.

### What minimal example or steps are needed to reproduce the bug?

```less
a {
	--x: pink // ;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "customSyntax": "postcss-less",
  "rules": {
    "@stylistic/less/declaration-block-trailing-semicolon": "never"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.less" --config myconfig.json --fix
```

### What did you expect to happen?

Nothing reported and nothing written: `less@4.9.1` compiles the file with `@@@` put behind that semicolon, so the semicolon is the comment's text.

### What actually happened?

The semicolon is reported and `--fix` writes `--x: pink // `. The `semicolon-in-comment` sweep has the same run over `// ;;`, `// ; ;`, a semicolon of code on the line under the comment, and a form feed in front of the semicolon.

### What is the cause?

Less reads a custom property's value with `permissiveValue`, which reads comments and entities first and hands whatever it cannot read that way to `$parseUntil`, which knows no `//`. So whether the semicolon is code turns on the value: behind `pink`, `[a]`, `f(x)` or `"s"` it is the comment's text, behind `pink !important`, `(a)`, `a=b`, `1 / 2` or `{a}` it is code. The branch closing #359 believed the flag for every custom property rather than carry that grammar. A Less variable and an at-rule part the same way, but `never` writes nothing behind either, since `requiresTrailingSemicolon` keeps the semicolon Less asks for there.

Telling the two readers apart is the same question as [#688](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/688) asks: how much of Less's expression grammar a guard carries.

Measured against `less@4.9.1` and `postcss-less@6.0.0`.
